src/StoppingCondition.jl

```diff
@@ -92,14 +103,189 @@ function BumpIterationLimit(p::DSProblem, i::Int)
 end
 
 
-#Precision limit
-struct MeshPrecisionStoppingCondition <: AbstractStoppingCondition end
+#Mesh size precision limit
+mutable struct MeshPrecisionStoppingCondition{T} <: AbstractStoppingCondition
+    cont_min_mesh_size::T
+    min_mesh_sizes::Vector{T}
+
+    function MeshPrecisionStoppingCondition{T}(min_mesh_size::Union{T, Nothing}=nothing) where T
+        c = new()
+
+        if min_mesh_size != nothing
+            c.cont_min_mesh_size = min_mesh_size
+        else
+            c.cont_min_mesh_size = get_min_mesh_size(T)
+        end
+
+        return c
+    end
+end
+
+function init_stoppingcondition(p::DSProblem, s::MeshPrecisionStoppingCondition)
+    s.min_mesh_sizes = map(δ_min -> δ_min > 0 ? δ_min : s.cont_min_mesh_size, p.config.mesh.δ_min)
+end
 
 StoppingConditionStatus(::MeshPrecisionStoppingCondition) = "Mesh Precision limit"
 
-CheckStoppingCondition(p::DSProblem, ::MeshPrecisionStoppingCondition) = GetMeshSize(p) > min_mesh_size(p)
-(GetMeshSize(p::DSProblem{T})::T) where T = p.config.mesh.Δᵐ
-min_mesh_size(::DSProblem{Float64}) = 1.1102230246251565e-16
-min_mesh_size(::DSProblem{T}) where T = eps(T)/2
+function CheckStoppingCondition(p::DSProblem, s::MeshPrecisionStoppingCondition)
+    if p.config.mesh.only_granular
+        p.config.mesh.l > -50
+    else
+        any(GetMeshSizeVector(p) .> s.min_mesh_sizes)
+    end
+end
+
+
+(GetMeshSizeVector(p::DSProblem{T})::Vector{T}) where T = p.config.mesh.δ
+
+get_min_mesh_size(::Type{Float64}) = 1.1102230246251565e-16
```

Comment:
Where does this constant come from?

Reply:
This originally defined in `min_mesh_size` which has been removed in this pull request. The constant was set as half the machine epsilon of a `Float64` (smallest increment between 1 and the next number represented by the type, described by the `eps` function). This was intended to define the smallest value a Float64 can represent, and therefore the minimum value for the mesh size.

However as floating points don't actually work in a linear fashion where this approach would work, I'd suggest changing this to a larger number that can be well represented by a `Float64` over a larger range. This was an oversight in the original code.